server/boardData.js

```diff
@@ -234,26 +225,43 @@ BoardData.prototype.validate = function validate(item, parent) {
 BoardData.load = function loadBoard(name) {
 	var boardData = new BoardData(name);
 	return new Promise((accept) => {
-		fs.readFile(boardData.file, function (err, data) {
-			try {
-				if (err) throw err;
-				boardData.board = JSON.parse(data);
-				for (id in boardData.board) boardData.validate(boardData.board[id]);
-				log('disk load', { 'board': boardData.name });
-			} catch (e) {
-				console.error("Unable to read history from " + boardData.file + ". The following error occured: " + e);
+		minioClient.getObject(bucketName, boardData.file, (err, dataStream) => {
+			const buffer = [];
+
+			function catchCallback(err, hack) {
+				console.error("Unable to read history from " + boardData.file + ". The following error occurred: " + err);
 				log('empty board creation', { 'board': boardData.name });
-				boardData.board = {}
+				boardData.board = {};
+				const data = Buffer.concat(buffer).toString("utf-8");
 				if (data) {
 					// There was an error loading the board, but some data was still read
 					var backup = backupFileName(boardData.file);
 					log("Writing the corrupted file to " + backup);
-					fs.writeFile(backup, data, function (err) {
+					minioClient.putObject(bucketName, backup, data, err => {
 						if (err) log("Error writing " + backup + ": " + err);
 					});
 				}
+				if (hack) {
+					accept(boardData);
+				}
+			}
+
+			if (err) {
+				catchCallback(err, true);
+				return;
 			}
-			accept(boardData);
+
+			dataStream.on("data", buffer.push);
+			dataStream.on("end", () => {
+				try {
+					boardData.board = Buffer.concat(buffer).toJSON(); // TODO: Might have to do string -> json first
+					for (id in boardData.board) boardData.validate(boardData.board[id]);
+					log('disk load', { 'board': boardData.name });
+				} catch (err) {
+					catchCallback(err, false);
+				}
+				accept(boardData);
+			});
```

Comment:
Maybe we could have a simple Persistence interface, that just asynchronously writes or read a value for a key, and have the backend (file or minio) chosen using an environment variable.